[ERROR] Not found cis.f5.com/ipamLabel

#### Setup Details
CIS Version : v2.11.0
Build: f5networks/k8s-bigip-ctlr:2.11.0
BIGIP Version: Big IP 16.1.2.1 Build 0.0.10 Point Release 1
AS3 Version: 3.36.0-6
Agent Mode: AS3
Orchestration: K8S
Orchestration Version: v1.20.15
Pool Mode: Nodeport

#### Description  
 When configuring VirtualServer from manifest and a LoadBalancer Service WITHOUT `cis.f5.com/ipamLabel` annotation, everything works properly, however misleading error appears in CIS logs

#### Steps To Reproduce
1) Configure LoadBalancer Service WITHOUT `cis.f5.com/ipamLabel` annotation
2) Configure VirtualServer or TransportServer for LoadBalancer Service

#### Expected Result 
No error in logs

#### Actual Result  
Everything works properly, but we get error in logs: 
```
[ERROR] Not found cis.f5.com/ipamLabel in <my service> Unable to process.
```

#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```  

#### Observations (if any)

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -12,7 +12,7 @@ Added Functionality
     * Ingress
         *
     * CRD
-        *
+        * Support for ExternalIP update of associated services of Type LB in TS CR
 
 Bug Fixes
 ````````````
@@ -21,6 +21,8 @@ Bug Fixes
 * `Issue 2797 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2797>`_: TLSProfile deletes a referenced SSL Profile when making changes or deleting a VS.
 * `Issue 2799 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2799>`_: VirtualServer deletes a referenced iRule when making changes or deleting a VS.
 * `Issue 2789 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2789>`_: AS3 Post delay - Not working as expected.
+* `Issue 2816 <https://github.com/F5Networks/k8s-bigip-ctlr/issues/2816>`_: Fix Error Not found cis.f5.com/ipamLabel
+
 
 2.12.0
 -------------

--- a/pkg/controller/informers_test.go
+++ b/pkg/controller/informers_test.go
@@ -411,6 +411,13 @@ var _ = Describe("Informers Tests", func() {
 		})
 
 		It("Service", func() {
+			// setting teem data
+			mockCtlr.TeemData = &teem.TeemsData{
+				ResourceType: teem.ResourceTypes{
+					IPAMSvcLB:   make(map[string]int),
+					IngressLink: make(map[string]int),
+				},
+			}
 			svc := test.NewService(
 				"SampleSVC",
 				"1",

--- a/pkg/controller/responseHandler.go
+++ b/pkg/controller/responseHandler.go
@@ -114,6 +114,17 @@ func (ctlr *Controller) responseHandler(respChan chan resourceStatusMeta) {
 						if _, found := rscUpdateMeta.failedTenants[partition]; !found {
 							// update the status for transport server as tenant posting is success
 							ctlr.updateTransportServerStatus(virtual, virtual.Status.VSAddress, "Ok")
+							// Update Corresponding Service Status of Type LB
+							var svcNamespace string
+							if virtual.Spec.Pool.ServiceNamespace != "" {
+								svcNamespace = virtual.Spec.Pool.ServiceNamespace
+							} else {
+								svcNamespace = virtual.Namespace
+							}
+							svc := ctlr.GetService(svcNamespace, virtual.Spec.Pool.Service)
+							if svc.Spec.Type == v1.ServiceTypeLoadBalancer {
+								ctlr.setLBServiceIngressStatus(svc, virtual.Status.VSAddress)
+							}
 						}
 					}
 				case Route:

--- a/pkg/controller/worker.go
+++ b/pkg/controller/worker.go
@@ -326,7 +326,6 @@ func (ctlr *Controller) processResources() bool {
 				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
-			break
 		}
 		if ctlr.initState {
 			break
@@ -395,7 +394,6 @@ func (ctlr *Controller) processResources() bool {
 				utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", key, err))
 				isRetryableError = true
 			}
-			break
 		}
 		switch ctlr.mode {
 		case OpenShiftMode:
@@ -1032,7 +1030,6 @@ func (ctlr *Controller) processVirtualServers(
 				log.Debugf("IP address requested for service: %s/%s", virtual.Namespace, virtual.Name)
 				return nil
 			}
-			virtual.Status.VSAddress = ip
 		}
 	} else {
 		if virtual.Spec.HostGroup == "" {
@@ -1055,6 +1052,8 @@ func (ctlr *Controller) processVirtualServers(
 			}
 		}
 	}
+	// Updating the virtual server IP Address status
+	virtual.Status.VSAddress = ip
 	// Depending on the ports defined, TLS type or Unsecured we will populate the resource config.
 	portStructs := ctlr.virtualPorts(virtual)
 
@@ -2039,7 +2038,6 @@ func (ctlr *Controller) processTransportServers(
 				log.Debugf("IP address requested for Transport Server: %s/%s", virtual.Namespace, virtual.Name)
 				return nil
 			}
-			virtual.Status.VSAddress = ip
 		}
 	} else {
 		if virtual.Spec.VirtualServerAddress == "" {
@@ -2047,7 +2045,8 @@ func (ctlr *Controller) processTransportServers(
 		}
 		ip = virtual.Spec.VirtualServerAddress
 	}
-
+	// Updating the virtual server IP Address status
+	virtual.Status.VSAddress = ip
 	var rsName string
 	if virtual.Spec.VirtualServerName != "" {
 		rsName = formatCustomVirtualServerName(
@@ -2246,18 +2245,18 @@ func (ctlr *Controller) processLBServices(
 	svc *v1.Service,
 	isSVCDeleted bool,
 ) error {
-	if ctlr.ipamCli == nil {
-		log.Error("IPAM is not enabled, Unable to process Services of Type LoadBalancer")
-		return nil
-	}
 
 	ipamLabel, ok := svc.Annotations[LBServiceIPAMLabelAnnotation]
 	if !ok {
-		log.Errorf("Not found %v in %v/%v. Unable to process.",
-			LBServiceIPAMLabelAnnotation,
+		log.Debugf("Service %v/%v does not have annotation %v, continuing.",
 			svc.Namespace,
 			svc.Name,
+			LBServiceIPAMLabelAnnotation,
 		)
+		return nil
+	}
+	if ctlr.ipamCli == nil {
+		log.Warningf("IPAM is not enabled, Unable to process Services of Type LoadBalancer")
 		return nil
 	}
 

